src/main/java/com/autotune/analyzer/recommendations/model/GenericRecommendationModel.java

```diff
@@ -0,0 +1,622 @@
+import com.autotune.analyzer.recommendations.RecommendationNotification;
+import com.autotune.analyzer.recommendations.utils.RecommendationUtils;
+import com.autotune.analyzer.services.UpdateRecommendations;
+import com.autotune.analyzer.utils.AnalyzerConstants;
+import com.autotune.common.data.metrics.AcceleratorMetricResult;
+import com.autotune.common.data.metrics.MetricAggregationInfoResults;
+import com.autotune.common.data.metrics.MetricResults;
+import com.autotune.common.data.result.IntervalResults;
+
+import com.autotune.common.utils.CommonUtils;
+import com.autotune.utils.KruizeConstants;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Timestamp;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.autotune.analyzer.recommendations.RecommendationConstants.RecommendationValueConstants.*;
+
+/**
+ * This is a generic class that implements RecommendationModel abstract class and provides the code for functions
+ * that are common across cost and performance recommendation model.
+ */
+
+public class GenericRecommendationModel implements RecommendationModel{
+
+    protected String name;
+    protected double modelMemoryPercentile;
+    protected double modelCPUPercentile;
+    protected double modelAcceleratorPercentile;
+
+    // constructor
+    public GenericRecommendationModel( String name, RecommendationTunables recommendationTunables) {
+        // null checks
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("Name cannot be null or empty");
+        }
+        if (recommendationTunables == null) {
+            throw new IllegalArgumentException("RecommendationTunables cannot be null");
+        }
+
+        this.name = name;
+        this.modelMemoryPercentile = recommendationTunables.getMemoryPercentile();
+        this.modelCPUPercentile = recommendationTunables.getCpuPercentile();
+        this.modelAcceleratorPercentile = recommendationTunables.getAcceleratorPercentile();
+
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateRecommendations.class);
+
+    @Override
+    public RecommendationConfigItem getCPURequestRecommendation(Map<Timestamp, IntervalResults> filteredResultsMap, ArrayList<RecommendationNotification> notifications) {
+        boolean setNotification = true;
+        if (null == notifications) {
+            LOGGER.error("Notifications Object passed is empty. The notifications are not sent as part of recommendation.");
+            setNotification = false;
+        }
+
+        RecommendationConfigItem recommendationConfigItem = null;
+        String format = "";
+        JSONArray cpuUsageList = getCPUUsageList(filteredResultsMap);
+        LOGGER.debug("cpuUsageList : {}", cpuUsageList);
+
+
+        // Extract "max" values from cpuUsageList
```

Comment:
Can we extract this part as well in a separate function since it's common in both `getCPURequestRecommendation()` and `getCPURequestRecommendationForNamespace()` ?